ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -409,11 +439,12 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
       "Expressing a non-consuming choice on a contract should not result in its archival") {
       context =>
         for {
-          ledger <- context.participant()
-          Vector(receiver, giver) <- ledger.allocateParties(2)
-          agreementFactory <- ledger.create(giver, AgreementFactory(receiver, giver))
-          _ <- ledger.exercise(receiver, agreementFactory.exerciseCreateAgreement)
-          transactions <- ledger.flatTransactions(receiver, giver)
+          Vector(alpha, beta) <- context.participants(2)
+          receiver <- alpha.allocateParty()
+          giver <- beta.allocateParty()
+          agreementFactory <- beta.create(giver, AgreementFactory(receiver, giver))
+          _ <- alpha.exercise(receiver, agreementFactory.exerciseCreateAgreement)
```

Comment:
Potential flakiness between `alpha` and `beta`.

Reply:
Since this tests tries do "prove non-existence" of something, it looks like it's less of a case of potential flakiness and more of a source of false positives, like in the case of https://github.com/digital-asset/daml/pull/2900#discussion_r324545682. Makes sense?

Reply:
👍

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2900/commits/4edfac1712a8deb54e5686d93d7840cf3e18979b